packages/notebook/src/widgetfactory.ts

```diff
@@ -94,10 +100,20 @@ export class NotebookWidgetFactory extends ABCWidgetFactory<
  * The namespace for `NotebookWidgetFactory` statics.
  */
 export namespace NotebookWidgetFactory {
+  export interface IToolbarItem {
+    name: string;
+    widget: Widget;
+  }
+
   /**
    * The options used to construct a `NotebookWidgetFactory`.
    */
   export interface IOptions extends DocumentRegistry.IWidgetFactoryOptions {
+    /**
+     * Array of toolbar items to be added to NotebookPanel.
+     */
+    toolbarItems?: IToolbarItem[];
```

Comment:
We should use an interface from `DocumentRegistry` here rather than redefining it locally.